lib/shared/addon/components/form-node-affinity-k8s/component.js

```diff
@@ -0,0 +1,125 @@
+ *    priority: preferred/required
+ *    term:preferred or required term
+ *   }
+ *
+ */
+  init(){
+    this._super(...arguments);
+    const allTerms = []
+    const addTerms = function(terms = [], priority){
+      terms.forEach((term) => {
+        allTerms.push({
+          priority,
+          term
+        })
+      })
+    }
+
+    addTerms(this.nodeAffinity?.preferredDuringSchedulingIgnoredDuringExecution, TERM_PRIORITY.PREFERRED)
+    addTerms(this.nodeAffinity?.requiredDuringSchedulingIgnoredDuringExecution?.nodeSelectorTerms, TERM_PRIORITY.REQUIRED)
+    set(this, 'allTerms', allTerms)
+  },
+
+  actions: {
+    addTerm(){
+      const neu = {
+        priority: TERM_PRIORITY.REQUIRED,
+        term:     { _id: randomStr() }
+      }
+
+      this.allTerms.push(neu)
+      get(this.nodeAffinity, 'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms').addObject(neu.term)
+      this.notifyPropertyChange('allTerms')
+      this.notifyPropertyChange('nodeAffinity')
+    },
+
+    removeTerm(term){
+      get(this, 'allTerms').removeObject(term)
+
+      if (term.priority === TERM_PRIORITY.REQUIRED){
+        const removeFrom = get(this.nodeAffinity, 'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms')
+
+        removeFrom.removeObject(term.term)
+      } else {
+        const removeFrom = get(this.nodeAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+
+        removeFrom.removeObject(term.term)
+      }
+      this.notifyPropertyChange('nodeAffinity')
+    },
+
+
+    typeChanged(term, old){
+      if (old === TERM_PRIORITY.REQUIRED){
+        const removeFrom = get(this.nodeAffinity, 'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms')
+        const addTo = get(this.nodeAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+
+        set(this.nodeAffinity, 'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms', this.removeFromSpec(term.term, removeFrom))
+        set(term, 'term', this.addWeight(term.term))
+
+        addTo.pushObject(term.term)
+      } else {
+        const removeFrom = get(this.nodeAffinity, 'preferredDuringSchedulingIgnoredDuringExecution')
+        const addTo = get(this.nodeAffinity, 'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms')
+
+        set(this.nodeAffinity, 'preferredDuringSchedulingIgnoredDuringExecution', this.removeFromSpec(term.term, removeFrom))
+        set(term, 'term', this.removeWeight(term.term))
+
+
+        addTo.pushObject(term.term)
+      }
+      this.notifyPropertyChange('nodeAffinity')
+    },
+
+  },
+
+  removeFromSpec: (term, array) => {
+    return array.filter((each) => each._id !== term._id)
+  },
+
+  addWeight: (term) => {
+    const out = {
+      preference: term,
+      _id:             term._id
+    }
+
+    delete out.preference._id
+
+    return out
+  },
+
+  removeWeight: (term) => {
+    const out = {
+      _id: term._id,
+      ...term.preference
+    }
+
+    return out
```

Comment:
Similarly, it looks like `out` isn't required
```suggestion
    return {
      _id: term._id,
      ...term.preference
    }
```